packages/login/src/components/LoginForm/LoginForm.js

```diff
@@ -12,6 +12,22 @@ import {
 import {Pages} from '../../types/Pages'
 
 export class LoginForm extends Component {
+  constructor(props) {
+    super(props)
+
+    this.setAutoFocus()
+  }
+
+  setAutoFocus() {
+    if (this.props.username) {
+      this.usernameFocus = false
+      this.passwordFocus = true
+    } else {
+      this.usernameFocus = true
+      this.passwordFocus = false
+    }
+  }
```

Comment:
still quite complicated and I believe it won't work in some cases, because `usernameFocus` and `passwordFocus` won't be updated when the props change. :)

you don't need a constructor or any other additional method in order to solve it. :) simply build the required conditions inside the render method.